### PR TITLE
[GIT] Automatic package publication

### PR DESCRIPTION
**Pull request by ReinderVosDeWael**
This PR will automatically upload our package to TestPyPi on creation of a new release. In the future we should redirect this towards the "real" PyPi, but for the current stage of this project this will allow us to 1) get ample experience to work out any kinks in the publication process, and 2) allow our beta testers to download/update their brainstat with pip.

Current version is available at: https://test.pypi.org/project/brainstat/
Current version can be installed with: `pip install -i https://test.pypi.org/simple/ brainstat`

Note: I cannot test the trigger on release until after merging - some modification to the CI trigger may have to be made once this is pushed into master.

### CODE REVIEW

**Comment by ReinderVosDeWael** [comment]
Returned this to a draft to just make it a little bit better. :)

**Comment by ReinderVosDeWael** [comment]
Alrighty - now with added pre-release and release creation. So here is how it _should_ work, but again I can only test this one it is on master.

- Whenever any new push is made to master, a pre-release version called "latest" will be updated with the current master commit. 
- When a version is tagged with a version name "v*" (e.g. "v0.0.1-alpha.2") then a release of that version is automatically generated. This same version will also be uploaded to TestPyPi.

In other words, the only thing we need to do to create a new BrainStat version and have it available everywhere is create a tag.

**Comment by PeerHerholz** [comment]
Awesome things happening here @ReinderVosDeWael! I guess we can't be sure ATM, but I assume this would work for the foreseeable future concerning package management things, etc. as e.g.  the way `pip` works/resolves dependencies addresses a later stage anyway once things are available via `pip` and `github` releases should work anyway.
